lib/TextDecoder-impl.js

```diff
@@ -0,0 +1,25 @@
+"use strict";
+
+const { TextDecoder } = global.TextDecoder ? global : require("text-encoding");
```

Comment:
Let's not depend on the system implementation. For a long time, the default Node.js build's `TextDecoder` doesn't support any encoding other than ASCII and Unicode, which we obviously wouldn't want. The whole goal of this package is to provide an independent implementation of it. `text-encoding` is archived and similar: we don't want to inherit any bugs from there.

Instead, implement the algorithms for these classes in the spec: https://encoding.spec.whatwg.org/#dom-textdecoder for the `TextDecoder` constructor for example. The auxiliary algorithms used in the algorithms, like "get an encoding", are all implemented by this package under `lib/whatwg-encoding.js`.